crates/core_simd/src/vector/ptr.rs

```diff
@@ -0,0 +1,55 @@
+//! Private implementation details of public gather/scatter APIs.
+use crate::SimdUsize;
+use core::mem;
+
+/// A vector of *const T.
+#[derive(Debug, Copy, Clone)]
+#[repr(simd)]
+pub(crate) struct SimdConst<T, const LANES: usize>([*const T; LANES]);
+
+impl<T, const LANES: usize> SimdConst<T, LANES>
```

Comment:
I would call this `SimdConstPtr`, I know it's only internal for now but I was pretty confused what it meant.